### ISSUE TITLE
Attempt to compress NaN value causes infinite loop + OOM

### ISSUE THREAD

**Issue by bryhoyt** [beenotung/compress-json#21]
If the input to the JavaScript version of `compress()` contains any NaN values, these are mistakenly interpreted as integers and sent to the function `int_to_s()`, which ends up looping infinitely and causing the interpreter to run out of memory.

For example: `compress({x: NaN, y: 0/0})`

It's understandable that compress-json wouldn't attempt to encode NaN values as they are not valid JSON, but it shouldn't crash the entire process. I would expect it to throw an error instead (similar to how `undefined` is [handled](https://github.com/beenotung/compress-json/issues/7)).

**Comment by beenotung** [comment]
Nice catch, `JSON.stringify` convert `NaN` and `Infinity` into `null`, I'll make it configurable to also silently convert it into null or throw error

**Comment by beenotung** [comment]
I've updated the behavior to silently convert NaN and Infinite into null by default, which matches the behavior of `JSON.stringify()`.

If you prefer it throw an error instead, you may config it as below:
```
import { config } from 'compress-json'

config.error_on_nan = true
config.error_on_infinite = true
```

The new version is released as compress-json@3.1.0 :rocket:

**Comment by beenotung** [comment]
Feel free to re-open the issue if the problem persist in your case.
